wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -440,21 +560,42 @@ int atmel_ecc_create_pms(int slotId, const uint8_t* peerKey, uint8_t* pms)
     /* free the ECDHE slot */
     atmel_ecc_free(slotIdEnc);
 #endif
-
     return ret;
 }
 
-int atmel_ecc_create_key(int slotId, byte* peerKey)
+static uint8_t getCurveType(int curve_id)
+{
+    switch(curve_id)
+    {
+        case ECC_SECP256R1: return TA_KEY_TYPE_ECCP256;
+        case ECC_SECP224R1: return TA_KEY_TYPE_ECCP224;
+        case ECC_SECP384R1: return TA_KEY_TYPE_ECCP384;
+        case ECC_SECP256K1: return TA_KEY_TYPE_SECP256K1;
+        case ECC_BRAINPOOLP256R1: return TA_KEY_TYPE_ECCBP256R1;
+        case ECC_CURVE_DEF: return TA_KEY_TYPE_ECCP256; /* default */
+        default: WOLFSSL_MSG("Curve not identified");
+            return MICROCHIP_INVALID_ECC;
+    }
+}
+
+int atmel_ecc_create_key(int slotId, int curve_id, byte* peerKey)
 {
     int ret;
 
     /* verify provided slotId */
     if (slotId == ATECC_INVALID_SLOT) {
         return WC_HW_WAIT_E;
     }
+#ifdef WOLFSSL_MICROCHIP_TA100
+
+    const uint8_t key_type = getCurveType(curve_id);
```

Comment:
Mixed variable declaration in code. Please make sure variables are declared at top of function or brace section. Thanks

Reply:
Fixed